lib/phoenix_live_view/channel.ex

```diff
@@ -577,7 +577,9 @@ defmodule Phoenix.LiveView.Channel do
   ## Mount
 
   defp mount(%{"session" => session_token} = params, from, phx_socket) do
-    check_version_match(params, phx_socket.endpoint)
+    unless Map.has_key?(phx_socket, :parent_pid) do
```

Comment:
phx_socket is not the same as lv_socket, so I think this will either always return true or return false. I think the best we can do for now is to check the version match only if the "vsn" parameter has been given.